playbooks/roles/airship-configure-worker/tasks/main.yml

```diff
@@ -1,95 +1 @@
-  failed_when:
-    - _enrollforucp.rc != 0
-    - _enrollforucp.stderr.find("already has a value (enabled)") == -1
-  tags:
-    - skip_ansible_lint
-
-- name: Enroll node for Airship UCP
-  command: "kubectl label nodes {{ ansible_hostname }} ucp-control-plane=enabled"
-  when: ('airship-ucp-workers' in group_names)
-  register: _enrollforucp
-  changed_when:
-    - _enrollforucp.rc == 0
-  failed_when:
-    - _enrollforucp.rc != 0
-    - _enrollforucp.stderr.find("already has a value (enabled)") == -1
-  tags:
-    - skip_ansible_lint
-
-- name: Enroll node for Airship Openstack control plane
-  command: "kubectl label nodes {{ ansible_hostname }} openstack-control-plane=enabled openvswitch=enabled"
-  when: ('airship-openstack-control-workers' in group_names)
-  register: _enrollforopenstackcp
-  changed_when:
-    - _enrollforopenstackcp.rc == 0
-  failed_when:
-    - _enrollforopenstackcp.rc != 0
-    - _enrollforopenstackcp.stderr.find("already has a value (enabled)") == -1
-  tags:
-    - skip_ansible_lint
-
-- name: Enroll node for Airship Openstack L3 agent
-  command: |
-    kubectl label nodes {{ ansible_hostname }} openstack-l3-agent=enabled
-  when: ('airship-openstack-l3-agent-workers' in group_names)
-  register: _enrollforopenstackl3agent
-  changed_when:
-    - _enrollforopenstackl3agent.rc == 0
-  failed_when:
-    - _enrollforopenstackl3agent.rc != 0
-    - _enrollforopenstackl3agent.stderr.find("already has a value (enabled)") == -1
-  tags:
-    - skip_ansible_lint
-
-- name: Enroll node for Airship Openstack compute plane
-  command: "kubectl label nodes {{ ansible_hostname }} openstack-compute-node=enabled openvswitch=enabled"
-  when: ('airship-openstack-compute-workers' in group_names)
-  register: _enrollforopenstackcomp
-  changed_when:
-    - _enrollforopenstackcomp.rc == 0
-  failed_when:
-    - _enrollforopenstackcomp.rc != 0
-    - _enrollforopenstackcomp.stderr.find("already has a value (enabled)") == -1
-  tags:
-    - skip_ansible_lint
-    - add_compute_node
-
-- name: Enroll node for Airship Openstack control plane primary class
-  command: |
-    kubectl label nodes {{ ansible_hostname }} openstack-helm-node-class=primary
-  when:
-    - ('airship-openstack-control-workers' in group_names)
-    - hostvars[inventory_hostname].primary is defined
-    - hostvars[inventory_hostname].primary == 'yes'
-  register: _enrollforopenstackcp
-  changed_when:
-    - _enrollforopenstackcp.rc == 0
-  failed_when:
-    - _enrollforopenstackcp.rc != 0
-    - _enrollforopenstackcp.stderr.find("already has a value (primary)") == -1
-  tags:
-    - skip_ansible_lint
-
-- name: Copy openstack apparmor profile into the workers
-  copy:
-    src: "{{ item }}"
-    dest: "/etc/apparmor.d/local/{{ item }}"
-  loop: "{{ apparmor_profiles }}"
-  register: apparmor_copy
-  notify: Reload apparmor profile
-  when:
-    - ('airship-openstack-control-workers' in group_names) or ('airship-openstack-compute-workers' in group_names)
+---
```

Comment:
Not sure, but can't we delete this file? Or is a `main.yml` a requirement for a role?

Reply:
I...dont know :o

Reply:
Seems like it:
`Roles expect files to be in certain directory names. Roles must include at least one of these directories, however it is perfectly fine to exclude any which are not being used. When in use, each directory must contain a main.yml file, which contains the relevant content`

from https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html

Reply:
thx